include/mxnet/ndarray.h

```diff
@@ -53,6 +67,27 @@ class AGNodeEntry {
 class AutogradRuntime;
 }  // namespace autograd
 
+// enum for storage types
+#define CSR_IND_PTR_TYPE mshadow::kInt32
+#define CSR_IDX_DTYPE mshadow::kInt32
+#define ROW_SPARSE_IDX_TYPE mshadow::kInt32
```

Comment:
yeah I'm thinking about default to 64bit for row_sparse::idx and csr::indices, and 32bit for csr::indptr. should be sufficient